### ISSUE TITLE
Whisper filter causes panic.

### ISSUE THREAD

**Issue by datajerk** [ethereum/go-ethereum#2439]
#### System information

Geth version:

```
Geth
Version: 1.3.5
Git Commit: 34b622a24853e1cf550238e8c8be6dfa4178cd35
Protocol Versions: [63 62 61]
Network Id: 1
Go Version: go1.6
OS: linux
GOPATH=
GOROOT=/tmp/go
```

OS & Version: `Linux 4.1.12-hypriotos-v7+ #2 SMP PREEMPT Tue Nov 3 19:44:55 UTC 2015 armv7l GNU/Linux`

Commit hash : `34b622a24853e1cf550238e8c8be6dfa4178cd35`
#### Expected behaviour

No Crash
#### Actual behaviour

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x205c48]

goroutine 23 [running]:
panic(0x830540, 0x11026030)
    /tmp/go/src/runtime/panic.go:464 +0x330
github.com/ethereum/go-ethereum/xeth.(*whisperFilter).insert(0x0, 0x12b92db8, 0x1, 0x1)
    /tmp/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/xeth/whisper_filter.go:72 +0x28
github.com/ethereum/go-ethereum/xeth.(*XEth).NewWhisperFilter.func1(0x1361bc80, 0x113eac00, 0x17c, 0x12b21800, 0x3, 0x115b4c60, 0x84, 0x57092f82, 0x0, 0xc8, ...)
    /tmp/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/xeth/xeth.go:647 +0x148
github.com/ethereum/go-ethereum/xeth.(*Whisper).Watch.func1(0x1361bc80)
    /tmp/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/xeth/whisper.go:107 +0x48
github.com/ethereum/go-ethereum/whisper.(*Whisper).Watch.func1(0x884908, 0x1361bc80)
    /tmp/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/whisper/whisper.go:142 +0x38
github.com/ethereum/go-ethereum/whisper.filterer.Trigger(0x0, 0x0, 0x0, 0x0, 0x12a3cee0, 0x127b8600, 0x884908, 0x1361bc80)
    /tmp/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/whisper/filter.go:131 +0x2c
github.com/ethereum/go-ethereum/whisper.(*filterer).Trigger(0x127b8620, 0x884908, 0x1361bc80)
    <autogenerated>:2 +0xa4
github.com/ethereum/go-ethereum/event/filter.(*Filters).loop(0x1108b640)
    /tmp/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/event/filter/filter.go:81 +0x330
created by github.com/ethereum/go-ethereum/event/filter.(*Filters).Start
    /tmp/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/event/filter/filter.go:49 +0x2c
```
#### Steps to reproduce the behaviour

Setup a whisper filter, poll it.
#### Backtrace

```
[backtrace]
```

**Comment by karalabe** [comment]
Does this happen on develop as well? We've completely removed the `xeth` package that this panic seems to originate from. And AFAIK a release candidate for 1.4 is imminent, so it may not really be worth to patch 1.3.x.

**Comment by datajerk** [comment]
No.  will test 1.4 when it releases.

**Comment by praveennagpal** [comment]
I am experiencing the same issue on 1.3.6.

panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x4215bcf]

goroutine 67 [running]:
panic(0x49bbc40, 0xc8200100a0)
	/usr/local/opt/go/libexec/src/runtime/panic.go:464 +0x3e6
github.com/ethereum/go-ethereum/xeth.(*whisperFilter).insert(0x0, 0xc8207f4bb8, 0x1, 0x1)
	/private/tmp/ethereum20160401-44941-n0f26s/build/_workspace/src/github.com/ethereum/go-ethereum/xeth/whisper_filter.go:72 +0x32f
github.com/ethereum/go-ethereum/xeth.(*XEth).NewWhisperFilter.func1(0xc837a10080, 0xc83936da40, 0x1e0, 0xc8391d40d3, 0x3, 0xc8391d40d0, 0x3, 0x570c5e47, 0x12c, 0xc83823e050, ...)
	/private/tmp/ethereum20160401-44941-n0f26s/build/_workspace/src/github.com/ethereum/go-ethereum/xeth/xeth.go:647 +0x162
github.com/ethereum/go-ethereum/xeth.(*Whisper).Watch.func1(0xc837a10080)
	/private/tmp/ethereum20160401-44941-n0f26s/build/_workspace/src/github.com/ethereum/go-ethereum/xeth/whisper.go:107 +0x62
github.com/ethereum/go-ethereum/whisper.(*Whisper).Watch.func1(0x4a62d20, 0xc837a10080)
	/private/tmp/ethereum20160401-44941-n0f26s/build/_workspace/src/github.com/ethereum/go-ethereum/whisper/whisper.go:142 +0x3d
github.com/ethereum/go-ethereum/whisper.filterer.Trigger(0x0, 0x0, 0x0, 0x0, 0xc834d03ac0, 0xc8389ed540, 0x4a62d20, 0xc837a10080)
	/private/tmp/ethereum20160401-44941-n0f26s/build/_workspace/src/github.com/ethereum/go-ethereum/whisper/filter.go:131 +0x30
github.com/ethereum/go-ethereum/whisper.(*filterer).Trigger(0xc837745500, 0x4a62d20, 0xc837a10080)
	<autogenerated>:2 +0xaa
github.com/ethereum/go-ethereum/event/filter.(*Filters).loop(0xc820194b20)
	/private/tmp/ethereum20160401-44941-n0f26s/build/_workspace/src/github.com/ethereum/go-ethereum/event/filter/filter.go:81 +0x424
created by github.com/ethereum/go-ethereum/event/filter.(*Filters).Start
	/private/tmp/ethereum20160401-44941-n0f26s/build/_workspace/src/github.com/ethereum/go-ethereum/event/filter/filter.go:49 +0x35

**Comment by obscuren** [comment]
If you are having issues with whisper please try out the develop branch. The parts that manage whisper and filter logic have been completely rewritten.

We won't be back porting any fixes to whisper on the `1.3` branch, this because `1.4` is due and because whisper is extremely experimental anyway.

**Comment by karalabe** [comment]
@datajerk Would be nice to test before release so we can fix it if it doesn't work. Otherwise it's just going to be delayed even further.

**Comment by hdiedrich** [comment]
Yes, what is ETA for 1.4 right now?

**Comment by obscuren** [comment]
Two Wooks™ ;-)

RC1 is due for this week.

**Comment by fjl** [comment]
This should be resolved with geth 1.5.0.
